hudi-common/src/main/java/org/apache/hudi/common/util/ParquetUtils.java

```diff
@@ -105,6 +121,111 @@
     return rowKeys;
   }
 
+  /**
+   * Fetch records from the given parquet file as {@link List} of {@link ParquetRowInfo}.
+   *
+   * @param filePath      The parquet file path.
+   * @param configuration configuration to build fs object
+   * @return {@link List} of {@link ParquetRowInfo}
+   */
+  public static List<ParquetRowInfo> fetchRecordKeyPartitionPathFromParquet(Configuration configuration, Path filePath,
+                                                                            String baseInstantTime,
+                                                                            String fileId) {
+    return fetchRecordKeyPartitionPathFromParquet(configuration, filePath, baseInstantTime, fileId, Collections.EMPTY_LIST);
+  }
+
+  /**
+   * Fetch records from the given parquet file as {@link List} of {@link ParquetRowInfo}. Filter for candidates if set.
+   *
+   * @param filePath      The parquet file path.
+   * @param configuration configuration to build fs object
+   * @return {@link List} of {@link ParquetRowInfo} matching candidateRecordKeys
+   */
+  static List<ParquetRowInfo> fetchRecordKeyPartitionPathFromParquet(Configuration configuration, Path filePath,
+                                                                     String baseInstantTime,
+                                                                     String fileId, List<String> candidatesToFilter) {
```

Comment:
we don't intend on doing an filtering right.. i.e this is never called? if so, can we just have the method above?